### ISSUE TITLE
The gdb package on darwin contains no binaries

### ISSUE THREAD

**Issue by Lucus16** [NixOS/nixpkgs#147953]
## Issue description

The `gdb` package on `aarch64-darwin` as built by hydra and also by my macbook does not contain any binaries or even a `bin` directory. From looking at the build log it seems to build a number of relevant libraries but never builds the `gdb` binary. The build still succeeds and leaves a package with just a `nix-support` and `share` directory. The logs say this:

```
checking for isl 0.15 or later... no
required isl version is 0.15 or later
*** This configuration is not supported in the following subdirectories:
     zlib readline gdbserver gdb sim
```

Adding `isl` (version 0.20) to the `buildInputs` still results in the above message.

### Steps to reproduce

Run `nix-build -A gdb` on a darwin machine. I don't know if this is `aarch64`-specific yet.

## Technical details

 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.1.0, macOS 12.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixpkgs-21.11pre333180.98747f27ecf"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by Lucus16** [comment]
I managed to add `isl` such that it is detected by the configure script but it still claims that `gdb` is not supported. Apparently that was not the reason.

**Comment by Lucus16** [comment]
It seems `gdb` does not yet support `aarch64-darwin`. Perhaps the build should be made to fail if it does not contain the `gdb` binary? The configure script seems to assume you might still be interested in various other parts of the package but as far as I can tell, none of them end up in the build output.

**Comment by starcraft66** [comment]
We can mark the package broken on `aarch64-darwin`.

**Comment by wegank** [comment]
Closed by #203373.
